### PR TITLE
Feature/indexer acc

### PR DESCRIPTION
**Pull request by binyebarwe**

### CODE REVIEW

File: programs/openbook-v2/src/instructions/create_open_orders_account.rs [hunk 1]
```diff
@@ -5,10 +5,13 @@ use anchor_lang::prelude::*;
...
     let mut account = ctx.accounts.open_orders_account.load_init()?;
-    let mut indexer = ctx.accounts.open_orders_indexer.load_mut()?;
-    indexer.created_counter += 1;
+    let indexer = &mut ctx.accounts.open_orders_indexer;
+    indexer
+        .addresses
+        .push(ctx.accounts.open_orders_account.key());
+    indexer.total_accounts += 1;
 
-    account.account_num = indexer.created_counter;
```
Comment (mschneider) [inline programs/openbook-v2/src/instructions/create_open_orders_account.rs:11]: nit: i like created_counter more as a term, it's more specific than total_accounts

File: programs/openbook-v2/src/state/open_orders_indexer.rs [hunk 2]
```diff
@@ -2,32 +2,23 @@ use anchor_lang::prelude::*;
...
-const_assert_eq!(size_of::<OpenOrdersIndexer>(), 96);
+const_assert_eq!(size_of::<OpenOrdersIndexer>(), 32);
 const_assert_eq!(size_of::<OpenOrdersIndexer>() % 8, 0);
 
 impl OpenOrdersIndexer {
-    /// Number of bytes needed for the account, including the discriminator
-    pub fn space() -> usize {
-        8 + size_of::<OpenOrdersIndexer>()
+    pub fn space(len: usize) -> usize {
+        8 + (4 + ((len + 1) * 32)) + 8
```
Comment (mschneider) [inline programs/openbook-v2/src/state/open_orders_indexer.rs:18]: should be 24 + len * 32

File: programs/openbook-v2/src/accounts_ix/create_open_orders_account.rs [hunk 3]
```diff
@@ -10,13 +10,16 @@ pub struct CreateOpenOrdersAccount<'info> {
     pub delegate_account: Option<UncheckedAccount<'info>>,
     #[account(
         mut,
-        has_one = owner,
-        has_one = market
+        seeds = [b"OpenOrdersIndexer".as_ref(), owner.key().as_ref()],
+        bump = open_orders_indexer.bump,
+        realloc = OpenOrdersIndexer::space(open_orders_indexer.addresses.len()+1),
+        realloc::payer = payer,
+        realloc::zero = false,
```
Comment (skrrb) [inline programs/openbook-v2/src/accounts_ix/create_open_orders_account.rs:17]: is`realloc::zero = false` safe? what happens if an user grows the account, then reduces it and finally grows it again?

Comment (binyebarwe) [reply programs/openbook-v2/src/accounts_ix/create_open_orders_account.rs:17]: The test is doind that right?

Comment (binyebarwe) [reply programs/openbook-v2/src/accounts_ix/create_open_orders_account.rs:17]: https://github.com/coral-xyz/anchor/pull/1943#issuecomment-1145985943

**Pull request merged**